### DISCUSSION TITLE
Nesting Components

### DISCUSSION THREAD

**Discussion by cmdobueno** [Z3d0X/filament-fabricator#36] [Q&A] [answered]
So I am curious if it is possible to next components.

I would like to have a grid based design on a page and allow this to be easily customized. Additionally, I want to define if a component would be full width or contained (container).

I might be missing some notes/documentation on this, or it simply may not be possible. 

Any advise would be awesome though!

**Comment by Z3d0X** [comment, marked as answer]
Here is a simple implementation of a grid block, with a toggle for full_width

```php
class GridBlock extends PageBlock
{
    public static function getBlockSchema(): Block
    {
        return Block::make('grid')
            ->schema([
                Toggle::make('full_width')
                    ->label('Full width')
                    ->default(false),

                TextInput::make('columns')
                    ->label('Columns')
                    ->integer()
                    ->required(),

                PageBuilder::make('blocks')
                    ->blocks([
                        MyBlock::getBlockSchema(),
                    ]),
            ]);
    }
}
```

```blade
@props([
    'blocks' => [],
    'columns' => 1,
    'full_width' => false,
])
<div {{ $attributes->class("px-4 py-4 md:py-8") }}>
    <div
        @class([
            'grid',
            'max-w-7xl mx-auto' => ! $full_width,
        ])
        style="grid-template-columns: repeat({{ $columns }}, minmax(0, 1fr));"
    >
        <x-filament-fabricator::page-blocks :blocks="$blocks" />
    </div>
</div>
```

**Comment by cmdobueno** [comment]
That is awesome thank you for the tip!!!

**Comment by mybouhssina** [comment]
@Z3d0X This is not working with the current version, it causes a 502 segfault error, am I missing something ?
I even tried it in a new project, same problem, no errors in the laravel.log

**Reply by mybouhssina** [reply]
a wordaround is to use Builder instead of PageBuilder in the example you provided, I'm not sure what's going on, couldn't figure out what's causing the Segfault, any clarification would be very appreciated !
